misc/review/MidiService.idl

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+// ============================================================================
+// This is part of the Windows MIDI Services App API and should be used
+// in your Windows application via an official binary distribution.
+// Further information: https://github.com/microsoft/MIDI/
+// ============================================================================
+
+#include "namespace_defines.h"
+#include "midl_defines.h"
+MIDI_IDL_IMPORT
+
+import "MidiServiceTransportPluginInfo.idl";
+import "MidiServiceMessageProcessingPluginInfo.idl";
+import "MidiServicePingResponseSummary.idl";
+
+import "MidiServiceSessionInfo.idl";
+
+import "MidiServiceLoopbackEndpointCreationResult.idl";
+import "MidiServiceLoopbackEndpointDefinition.idl";
+
+import "IMidiServiceTransportPluginConfiguration.idl";
+import "IMidiServiceMessageProcessingPluginConfiguration.idl";
+import "MidiServiceConfigurationResponse.idl";
+
+namespace MIDI_MIDL_NAMESPACE
+{
+    MIDI_API_CONTRACT_v1
+    [default_interface]
+    static runtimeclass MidiService
+    {
+        static MidiServicePingResponseSummary PingService(UInt8 pingCount);
+        static MidiServicePingResponseSummary PingService(UInt8 pingCount, UInt32 timeoutMilliseconds);
+
+        static IVector<MidiServiceTransportPluginInfo> GetInstalledTransportPlugins();
+        static IVector<MidiServiceMessageProcessingPluginInfo> GetInstalledMessageProcessingPlugins();
+
+        static Boolean IsAvailable();
```

Comment:
I've actually changed this to EnsureAvailable() because the service needs to be demand start. So we trigger the start through this method and then do the checks that were happening previously in IsAvailable. It's a required steps for apps to use now that we're shipping this way, and does need to block. Note our partners have made it clear that they do not like WinRT Async methods, and given that nothing works until this is done at least once per boot, it seems blocking should be ok here. Thoughts?

Reply:
Sync is fine. We like sync now.